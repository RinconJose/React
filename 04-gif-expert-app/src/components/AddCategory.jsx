import { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('One Punch')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmitForm = (event) => {
        event.preventDefault();
        const cleanValue = inputValue.trim();
        if(cleanValue.length <= 1) return;
        // setCategories( categories => [inputValue, ...categories] )
        setInputValue('')
        onNewCategory(cleanValue)
    }

    return (
        <form onSubmit={onSubmitForm} aria-label='form'>
            <input
                type="text"
                placeholder="Buscar gift"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}