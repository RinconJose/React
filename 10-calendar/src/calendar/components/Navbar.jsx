import { useAuthStore } from '../../hooks'


export const Navbar = () => {

    const {startLogout, user} = useAuthStore();

    return (
        <nav className="navbar navbar-dark bg-dark mb-4 px-4 justify-content-between">
            <span className="navbar-brand d-flex align-self-center p-0">
                <i className="fas fa-calendar-alt m-auto"></i>
                &nbsp;
                <span className="m-auto">{user.name}</span>
            </span>
            <button
                className="btn btn-outline-danger"
                onClick={startLogout}
            >
                <i className="fas fa-sign-out-alt"></i>
                &nbsp;
                <span>Salir</span>
            </button>
        </nav>
    )
}
