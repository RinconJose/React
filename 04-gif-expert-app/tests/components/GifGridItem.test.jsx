import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

describe('Pruebas en GifGridItem', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Titulo y url obligatorias', () => {
        const {container} = render(<GifGridItem title={title} url={url} />)
        expect( container ).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen conel url y alt indicado', () => {
        render(<GifGridItem title={title} url={url} />)
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe(url);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy()
    })
})