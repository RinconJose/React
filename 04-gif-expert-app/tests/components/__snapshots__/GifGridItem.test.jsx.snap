// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Titulo y url obligatorias 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
